utilities/cache_utils.py

```
@@ -0,0 +1,469 @@
+        logging.error(f"Failed to save {cache_type} from disk cache: {e}")
+        print(f"Failed to save {cache_type} from disk cache: {e}")
+        print(traceback.format_exc())
+
+def load_from_disk(cache_key, cache_type, device: torch.device):
+    try:
+        if cache_type == "prefix":
+            cache_dir = get_prefix_cache_dir()
+        elif cache_type == "embeds":
+            cache_dir = get_embed_cache_dir()
+        else:
+            raise ValueError(f"Unknown cache type: {cache_type}")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)